embassy-executor-macros/src/macros/main.rs

```diff
@@ -13,45 +13,58 @@ enum Flavor {
     Wasm,
 }
 
+pub(crate) enum Entry {
+    NotRequired,
+    Required,
+    Some(&'static str),
```

Comment:
this is somewhat strange. What do the 3 cases mean?

Could you instead leave `default_entry` unchanged and add another field `requires_executor` to `Arch`?

Reply:
I could but it would complicate the pattern match significantly.

Reply:
Renamed to `None`, `Required` and `Default`, which might be a little clearer. I can redo it with a boolean if you'd like though.

Reply:
Actually I'll just try it and see if it seems ok

Reply:
why? there's no need to touch the `entry` code at all. 

If no `arch-*` feature is enabled and `entry="...."` is not specified in the macro, just don't apply any entry macro. We shouldn't error in this case.